doc/python-interface/index.md

```diff
@@ -0,0 +1,11 @@
+(python_spglib)=
```

Comment:
Let's avoid explicit target definitions. We can better reference documents (`python-interface/index.md`) or auto-generated links `python-interface/index.md#spglib-for-python`)

Reply:
I feel references by auto-generated links are fragile, but linking by file names would be good.

Reply:
Yes, they are more fragile, but they are more navigable since the IDE also includes the same auto-generated target, and these should be picked up the `linkcheck`. Definitely though, should avoid using global targets.

Reply:
Currently, a lot of docstrings refer to explicit target definitions. Since they are rst, I think we cannot fully migrate the implicit target definitions for now. Replacing the existing targets will require patience (not in this PR).